chore(sdk): Drop Python 3.9 support

aniketpati1121 opened the pull request:
This PR removes support for Python 3.9 across the Kubeflow Pipelines SDK. 
Python 3.9 is reaching end-of-life in October 2025, and this change ensures 
our SDK dependencies remain up-to-date and secure. #11922 

Changes include:
- Updated CI/CD workflows to remove Python 3.9 from test matrices.
- Updated setup.py / pyproject.toml to require Python >=3.10.
- Updated Dockerfiles and other environment configurations.
- Updated documentation and README to reflect the new supported Python versions.

HumairAK commented on backend/src/apiserver/resource/resource_manager_util_test.go:
```diff
@@ -92,7 +92,7 @@ status:
       boundaryID: resubmit-hl9ft
       children:
-      - resubmit-hl9ft-3895868335
+      - resubmit-hl9ft-3.105868335
```
undo this, same with [this](https://github.com/kubeflow/pipelines/pull/12243/files#diff-629f33eea4675d8431f5c7dc7605130cb5493f68cd558c15b8a0b09607cc598dR105) and [this](https://github.com/kubeflow/pipelines/pull/12243/files#diff-629f33eea4675d8431f5c7dc7605130cb5493f68cd558c15b8a0b09607cc598dR109)

HumairAK commented on backend/src/apiserver/server/test/xgboost_sample_pipeline.yaml:
```diff
@@ -257,7 +257,7 @@ deploymentSpec:
         - '{{$.inputs.parameters[''limit'']}}'
         - '{{$.inputs.parameters[''format'']}}'
-        image: byrnedo/alpine-curl@sha256:548379d0a4a0c08b9e55d9d87a592b7d35d9ab3037f4936f5ccd09d0b625a342
+        image: byrnedo/alpine-curl@sha256:5483.10d0a4a0c08b9e55d9d87a592b7d35d9ab3037f4936f5ccd09d0b625a342
```
undo this

HumairAK commented on backend/src/apiserver/server/api_converter_test.go:
```diff
@@ -3333,7 +3333,7 @@ func Test_toModelTasks_wf(t *testing.T) {
 			RunId:             "run1_uid_true",
 			CreatedTimestamp:  -62135596800,
-			StartedTimestamp:  1675734919,
+			StartedTimestamp:  167573.1019,
```
undo this

zazulam reviewed (changes requested):
Nice work! Just noticed a few spots either for updating some surrounding context or additional changes that can be undone.

zazulam commented:
@aniketpati1121 can you also rebase with master after you make the requested changes so the CI can execute

aniketpati1121 commented:
> @aniketpati1121 can you also rebase with master after you make the requested changes so the CI can execute

Hi @zazulam,

I’ve rebased the branch with master and reverted the Python version back to 3.9 as requested.
All the changes you suggested for undoing outdated updates have been applied.
The branch is now up to date and ready for CI execution.

Thanks a lot for your guidance and support!

– Aniket

zazulam commented:
@aniketpati1121 please update your master branch and rebase your PR branch so it is not marked out-of-date
<img width="430" height="70" alt="image" src="https://github.com/user-attachments/assets/ce581c96-8bb0-498d-a8df-972dc34e87cb" />

aniketpati1121 commented:
> @aniketpati1121 please update your master branch and rebase your PR branch so it is not marked out-of-date <img alt="image" width="430" height="70" src="https://private-user-images.githubusercontent.com/13690060/491557499-ce581c96-8bb0-498d-a8df-972dc34e87cb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTgyODM1NTgsIm5iZiI6MTc1ODI4MzI1OCwicGF0aCI6Ii8xMzY5MDA2MC80OTE1NTc0OTktY2U1ODFjOTYtOGJiMC00OThkLWE4ZGYtOTcyZGMzNGU4N2NiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTE5VDEyMDA1OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2MThlOTFmNzcyZDc1NmZiN2RlY2E2YmIzYjExN2E5ZDc1Yzc4YzY1ZTgwMWMwYjQ5ZGVhMjA5ZmIxYThhMmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.PAeEyyLx40oOiTd7_HKpwaVyHl1Akq5KPZdKYuBBxD0">

Done

zazulam commented:
@aniketpati1121 I found this in one of the failed workflows
> `2025-09-30T09:29:37.165176216Z ERROR: Package 'kfp' requires a different Python: 3.9.23 not in '>=3.10.0'`

Please ensure that any references to python:3.9 images are updated in the sdk. You might have to re-compile the pipeline yamls that are used for testing so that the image specified is using python:3.10 for the components.
